roachtest: asyncpg failed

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.asyncpg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18554559?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18554559?buildTab=artifacts#/asyncpg) on master @ [89d0fdd3a2a8c35ade62271f4eb63d10dcfe6b60](https://github.com/cockroachdb/cockroach/commits/89d0fdd3a2a8c35ade62271f4eb63d10dcfe6b60):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.1.0-alpha.3-dev-89d0fdd3a2a8c35ade62271f4eb63d10dcfe6b60
Tests run against asyncpg v0.24.0
212 Total Tests Run
165 tests passed
47 tests failed
34 tests skipped
15 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test_codecs.TestCodecs.test_enum (unexpected)
For a full summary look at the asyncpg artifacts 
An updated blocklist (asyncpgBlocklist) is available in the artifacts' asyncpg log
test artifacts and logs in: /artifacts/asyncpg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=16</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18554559/asyncpg/1737443942578/1737444213117)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*asyncpg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46679

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.asyncpg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18554570?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18554570?buildTab=artifacts#/asyncpg) on master @ [89d0fdd3a2a8c35ade62271f4eb63d10dcfe6b60](https://github.com/cockroachdb/cockroach/commits/89d0fdd3a2a8c35ade62271f4eb63d10dcfe6b60):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.1.0-alpha.3-dev-89d0fdd3a2a8c35ade62271f4eb63d10dcfe6b60
Tests run against asyncpg v0.24.0
212 Total Tests Run
165 tests passed
47 tests failed
34 tests skipped
15 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test_codecs.TestCodecs.test_enum (unexpected)
For a full summary look at the asyncpg artifacts 
An updated blocklist (asyncpgBlocklist) is available in the artifacts' asyncpg log
test artifacts and logs in: /artifacts/asyncpg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=16</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*asyncpg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.asyncpg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18565590?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18565590?buildTab=artifacts#/asyncpg) on master @ [93fb203a469911c4a3ca7fb79f9a94adcb38689d](https://github.com/cockroachdb/cockroach/commits/93fb203a469911c4a3ca7fb79f9a94adcb38689d):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-93fb203a469911c4a3ca7fb79f9a94adcb38689d
Tests run against asyncpg v0.24.0
212 Total Tests Run
165 tests passed
47 tests failed
34 tests skipped
15 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test_codecs.TestCodecs.test_enum (unexpected)
For a full summary look at the asyncpg artifacts 
An updated blocklist (asyncpgBlocklist) is available in the artifacts' asyncpg log
test artifacts and logs in: /artifacts/asyncpg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=16</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18565590/asyncpg/1737528302480/1737528584754)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139472 roachtest: asyncpg failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*asyncpg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.asyncpg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18580451?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18580451?buildTab=artifacts#/asyncpg) on master @ [bf4dd7a00b611cbc1d428dd005bd2ec13e78ce1f](https://github.com/cockroachdb/cockroach/commits/bf4dd7a00b611cbc1d428dd005bd2ec13e78ce1f):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-bf4dd7a00b611cbc1d428dd005bd2ec13e78ce1f
Tests run against asyncpg v0.24.0
212 Total Tests Run
165 tests passed
47 tests failed
34 tests skipped
15 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test_codecs.TestCodecs.test_enum (unexpected)
For a full summary look at the asyncpg artifacts 
An updated blocklist (asyncpgBlocklist) is available in the artifacts' asyncpg log
test artifacts and logs in: /artifacts/asyncpg/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=16</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=leader</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139546 roachtest: asyncpg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.11-rc release-blocker]
- #139545 roachtest: asyncpg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2 release-blocker]
- #139544 roachtest: asyncpg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.9-rc release-blocker]
- #139543 roachtest: asyncpg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1 release-blocker]
- #139541 roachtest: asyncpg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3 release-blocker]
- #139472 roachtest: asyncpg failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*asyncpg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.asyncpg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18580440?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18580440?buildTab=artifacts#/asyncpg) on master @ [bf4dd7a00b611cbc1d428dd005bd2ec13e78ce1f](https://github.com/cockroachdb/cockroach/commits/bf4dd7a00b611cbc1d428dd005bd2ec13e78ce1f):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-bf4dd7a00b611cbc1d428dd005bd2ec13e78ce1f
Tests run against asyncpg v0.24.0
212 Total Tests Run
165 tests passed
47 tests failed
34 tests skipped
15 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test_codecs.TestCodecs.test_enum (unexpected)
For a full summary look at the asyncpg artifacts 
An updated blocklist (asyncpgBlocklist) is available in the artifacts' asyncpg log
test artifacts and logs in: /artifacts/asyncpg/cpu_arch=arm64/run_1
```

Parameters:
 - <code>arch=arm64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=16</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=false</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18580440/asyncpg/1737617416219/1737617743371)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139546 roachtest: asyncpg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1.11-rc release-blocker]
- #139545 roachtest: asyncpg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2 release-blocker]
- #139544 roachtest: asyncpg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.9-rc release-blocker]
- #139543 roachtest: asyncpg failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1 release-blocker]
- #139541 roachtest: asyncpg failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3 release-blocker]
- #139472 roachtest: asyncpg failed [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*asyncpg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/asyncpg.go
+++ b/pkg/cmd/roachtest/tests/asyncpg.go
@@ -18,11 +18,14 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
 )
 
+var asyncpgWarningArgs = "ignore::ResourceWarning"
+
 var asyncpgRunTestCmd = fmt.Sprintf(`
 source venv/bin/activate &&
 cd /mnt/data1/asyncpg &&
-PGPORT={pgport:1} PGHOST=localhost PGUSER=%s PGPASSWORD=%s PGSSLROOTCERT=$HOME/%s/ca.crt PGSSLMODE=require PGDATABASE=defaultdb python3 setup.py test > asyncpg.stdout
-`, install.DefaultUser, install.DefaultPassword, install.CockroachNodeCertsDir)
+PGPORT={pgport:1} PGHOST=localhost PGUSER=%s PGPASSWORD=%s PGSSLROOTCERT=$HOME/%s/ca.crt PGSSLMODE=require PGDATABASE=defaultdb python3 -W%q setup.py test > asyncpg.stdout
+`, install.DefaultUser, install.DefaultPassword, install.CockroachNodeCertsDir, asyncpgWarningArgs,
+)
 
 var asyncpgReleaseTagRegex = regexp.MustCompile(`^(?P<major>v\d+)\.(?P<minor>\d+)\.(?P<point>\d+)$`)
 

